.github/workflows/plan.yml

```diff
@@ -2,14 +2,53 @@ name: terraform
 on:
   pull_request:
     branches: [main]
+  pull_request_target:
+    branches: [main]
 
 jobs:
-  plan:
+  validate:
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      EMAIL: ${{ github.event.pusher.email }}
+      PR_REPO_FULL_NAME: ${{ github.event.pull_request.head.repo.full_name }}
     runs-on: ubuntu-latest
+    outputs:
+      valid_workflow: ${{ steps.validate_workflow.outputs.valid_workflow }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: psf/black@ae2c0758c9e61a385df9700dc9c231bf54887041 #v22.3.0
+      - name: Install and configure Poetry
+        uses: snok/install-poetry@2bf112a0f6979928eb6b011f39700db589c5961e #v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          version: 1.1.13
+      - name: Install library
+        run: |
+          cd scripts
+          poetry install
+      - name: Run tests
+        run: |
+          cd scripts
+          source .venv/bin/activate
```

Comment:
Do you need to `source` ? `poetry run` should automatically use the virtual environment it was configured with?